src/Sulu/Bundle/AdminBundle/Metadata/SchemaMetadata/SelectionPropertyMetadataMapper.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) Sulu GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\AdminBundle\Metadata\SchemaMetadata;
+
+use Sulu\Component\Content\Metadata\PropertyMetadata as ContentPropertyMetadata;
+
+class SelectionPropertyMetadataMapper implements PropertyMetadataMapperInterface
+{
+    public function mapPropertyMetadata(ContentPropertyMetadata $propertyMetadata): PropertyMetadata
+    {
+        $mandatory = $propertyMetadata->isRequired();
+        $minItems = $propertyMetadata->getParameter('min', true)['value'] ?? null;
```

Comment:
Currently, there is still a problem here ... if a property has `mandatory: false` and `min: 2`, I would expect to be able to either select `0` or `>= 2` items ... But unfortunately that's not possible with only json-schema, at least I don't know how. Probably we would have to change the `MultiSelection` to return `null` instead of an empty array, if no items are selected, but that's just a guess.

Reply:
@danrot WDYT about that?

Reply:
I don't agree on that (my personal opinion) ... /cc @sulu/core-team

Reply:
not sure if there is a correct behaviour, think i would go with whats easier to implement 😅

Reply:
Let's compare it with the `minLength` attribute of the `text_line`: if `mandatory: false` and `minLength: 2`, null or empty string will pass validation, one character will fail and all above to characters will pass again ... So i guess this is the correct behaviour ... If the value of the selection would be null, validation would also pass, if the field is not required but has `minItems: 2`

Reply:
That's not what I would expect, even if it is currently implemented like that in the `text_line`. As said, if I put a `min` attribute on it, I am doing that because I want there to be always at least to characters. And no characters is also less than 2, so the constraint I put on this element is not fulfilled, and therefore should not validate IMO.

Reply:
To summarize that - there are two options:

1. All other constraints than `mandatory` (like `minLength`) do always apply, even if a field is optional and has no value.
Using this option, it's not possible to save a field with `mandatory: false` and `min: 2`, even if the field is optional (IMO optional fields don't need to be filled, no matter what constraints they have)
2. All other constraints than `mandatory` only apply, if a field has a value.
This option has the advantage to say: "The amount of teasers in a `text-with-teasers` block type can either be `0` or exactly `3`."

I think this is a use case that's needed quite often.
@sulu/core-team please share your opinions

Reply:
@luca-rath In symfony "null" and "empty strings" are still valid for example constraints like:

 - https://symfony.com/doc/current/reference/constraints/Length.html 
 - https://symfony.com/doc/current/reference/constraints/Email.html
 - and many others

See there note:

> As with most of the other constraints, null and empty strings are considered valid values. This is to allow them to be optional values. If the value is mandatory, a common solution is to combine this constraint with NotBlank.

And I think we should go the same way, as it is the most flexible. So if something is not mandatory it can be still valid but I can tell that if it is set it need to have at least 10 letters in it.

Reply:
Think its a good idea to make this consistent with symfony constraints if possible 👍

Reply:
For JSON Schema when making a string also null able it would also work:

<details>

<summary>Example</summary>

```json
{
  "$schema": "http://json-schema.org/draft-04/schema#",
  "type": "object",
  "properties": {
    "test": {
      "type": ["string", "null"],
      "minLength": 2,
      "maxLength": 5
    },
    "test2": {
      "type": ["string", "null"],
      "minLength": 2,
      "maxLength": 5
    }
  },
  "required": [
    "test"
  ]
}
```

Example invalid:

```json
{
	"test": "invalid",
	"test2": "i",
}
```

```json
{
	"test": "valid",
	"test2": "12",
}
```

```json
{
	"test": "valid",
	"test2": null,
}
```

</details>

Reply:
Well, if Symfony also implements it like this, it is OK for me, although it still feels weird to me.

Reply:
okay, i think we got a decision ... will implement option 2